climada/hazard/test/test_tc_tracks.py

```diff
@@ -480,18 +481,23 @@ def test_interp_track_pass(self):
 
         # test some "generic floats"
         for time_step_h in [0.6663545049172093, 2.509374054925788, 8.175754471661111]:
-            tc_track = tc.TCTracks()
-            tc_track.read_processed_ibtracs_csv(TEST_TRACK)
-            tc_track.equal_timestep(time_step_h=time_step_h)
-            self.assertTrue(np.all(tc_track.data[0].time_step == time_step_h))
+            # artifically create data that doesn't start at full hour
+            for loffset in [0, 22, 30]:
+                tc_track = tc.TCTracks()
+                tc_track.read_processed_ibtracs_csv(TEST_TRACK)
+                tc_track.data[0].time.values[:] += np.timedelta64(loffset, "m")
+                tc_track.equal_timestep(time_step_h=time_step_h)
+                np.testing.assert_array_equal(tc_track.data[0].time_step, time_step_h)
+                self.assertTrue(np.isfinite(tc_track.data[0].central_pressure.values).all())
 
         tc_track = tc.TCTracks()
         tc_track.read_processed_ibtracs_csv(TEST_TRACK)
         tc_track.equal_timestep(time_step_h=0.16667)
 
-        self.assertEqual(tc_track.data[0].time.size, 1333)
+        self.assertEqual(tc_track.data[0].time.size, 1332)
```

Comment:
The first time step used to be a NaN produced by `equal_timestep`. With this PR, the first time step is discarded because it's outside of the original time range.